services/user/Identity/service/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+    pub struct Attestation {
+        /// The attesting account / the issuer
+        pub attester: AccountNumber,
+
+        /// creation/issue time
+        pub issued: psibase::TimePointSec,
+
+        /// The credential subject, in this case, the subject/attestee
+        pub subject: AccountNumber,
+
+        pub value: u8,
+    }
+
+    impl Attestation {
+        #[primary_key]
+        fn by_attestee(&self) -> (AccountNumber, AccountNumber) {
+            (self.subject, self.attester)
+        }
+
+        #[secondary_key(1)]
+        fn by_attester(&self) -> (AccountNumber, AccountNumber) {
+            (self.attester, self.subject)
+        }
+    }
+
+    impl fmt::Display for Attestation {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            writeln!(
+                f,
+                "--Attestation--\n  {}: attester\n  {}: subject\n  {}: value\n  {}: issued",
+                self.attester, self.subject, self.value, self.issued.seconds
+            )
+        }
+    }
+
+    #[table(name = "AttestationStatsTable", index = 1)]
+    #[derive(Fracpack, Reflect, Serialize, Deserialize, SimpleObject, Debug, Clone)]
+    pub struct AttestationStats {
+        /// The credential subject, in this case, the subject/attestee
+        #[primary_key]
+        pub subject: AccountNumber,
+
+        // % high conf + # unique attestations will give an approximation of a Google Review for a user
+        // pub perc_high_conf: u8,
+        pub num_high_conf_attestations: u16,
+
+        pub unique_attesters: u16,
+
+        // freshness indicator
+        pub most_recent_attestation: TimePointSec,
+    }
+
+    impl fmt::Display for AttestationStats {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            writeln!(
+                f,
+                "--AttestationStats--\n{}\t: subject\n{} :# unique attesters\n{}\t: % high conf\n{} :most recent attestation",
+                self.subject,
+                self.unique_attesters,
+                self.num_high_conf_attestations,
+                self.most_recent_attestation.seconds
+            )
+        }
+    }
+
+    impl Default for AttestationStats {
+        fn default() -> Self {
+            AttestationStats {
+                subject: AccountNumber::from(0),
+                num_high_conf_attestations: 0,
+                unique_attesters: 0,
+                most_recent_attestation: TimePointSec::from(0),
+            }
+        }
+    }
+
+    #[table(record = "WebContentRow", index = 2)]
+    struct WebContentTable;
+
+    #[action]
+    pub fn attest(subject: String, value: u8) {
+        check(value <= 100, "bad confidence score");
+        let attester = get_sender();
+        let issued = transact::Wrapper::call().currentBlock().time;
+
+        let attestation_table = AttestationTable::new();
+
+        let subj_acct = match AccountNumber::from_str(&subject.clone()) {
```

Comment:
I assume you mean that a string will automatically be converted to an AccountNumber if that's the arg type?
Then I need to verify the account exists anyway.
If we take the arg as a string, then we know we don't have a round-tripping problem. I first check that it round trips. Then I see if it exists.

does it achieve all the same stuff if the arg is AccountNumber and all I do is see if it exists?

Reply:
You don't need to check roundtripping, you just need to check that the account exists (which implies roundtripping because that's checked on account creation).

Reply:
> I assume you mean that a string will automatically be converted to an AccountNumber if that's the arg type?

I mean that the type of the thing passed over the wire can be an AccountNumber already, no need for conversion within the service